test/integration/nodeagent/lease/lease_test.go

```diff
@@ -0,0 +1,158 @@
+			}()
+
+			DeferCleanup(func() {
+				By("Stop manager")
+				mgrCancel()
+			})
+
+			node = &corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:   nodeName,
+					Labels: map[string]string{testID: testRunID},
+				},
+			}
+
+			By("Create Node")
+			Expect(testClient.Create(ctx, node)).To(Succeed())
+			DeferCleanup(func() {
+				By("Delete Node")
+				Expect(client.IgnoreNotFound(testClient.Delete(ctx, node))).To(Succeed())
+				By("Delete Lease")
+				lease := &coordinationv1.Lease{ObjectMeta: metav1.ObjectMeta{Namespace: testNamespace.Name, Name: "gardener-node-agent-" + nodeName}}
+				Expect(client.IgnoreNotFound(testClient.Delete(ctx, lease))).To(Succeed())
+			})
+		})
+
+		It("Reconcile should create lease", func() {
+			lease := &coordinationv1.Lease{}
+			Eventually(func() error {
+				return testClient.Get(ctx, types.NamespacedName{Namespace: testNamespace.Name, Name: "gardener-node-agent-" + nodeName}, lease)
+			}).Should(Succeed())
+			validateOwnerReference(lease, node)
+		})
+
+		It("Reconcile should update lease time", func() {
+			lease := &coordinationv1.Lease{}
+			Eventually(func() error {
+				return testClient.Get(ctx, types.NamespacedName{Namespace: testNamespace.Name, Name: "gardener-node-agent-" + nodeName}, lease)
+			}).Should(Succeed())
+			validateOwnerReference(lease, node)
+			oldRenewTime := lease.Spec.RenewTime
+			fakeclock.SetTime(fakeclock.Now().Add(1 * time.Minute))
+
+			Eventually(func() bool {
+				Expect(testClient.Get(ctx, types.NamespacedName{Namespace: testNamespace.Name, Name: "gardener-node-agent-" + nodeName}, lease)).To(Succeed())
+				return lease.Spec.RenewTime.After(oldRenewTime.Time)
+			}).Should(BeTrue())
+			validateOwnerReference(lease, node)
+		})
+
+		It("Reconcile should not update lease time if no node is present", func() {
+			lease := &coordinationv1.Lease{}
+			Eventually(func() error {
+				return testClient.Get(ctx, types.NamespacedName{Namespace: testNamespace.Name, Name: "gardener-node-agent-" + nodeName}, lease)
+			}).Should(Succeed())
+			validateOwnerReference(lease, node)
+			oldRenewTime := lease.Spec.RenewTime
+			fakeclock.SetTime(fakeclock.Now().Add(1 * time.Minute))
+
+			Expect(testClient.Delete(ctx, node)).To(Succeed())
+			Eventually(func() error {
+				return testClient.Get(ctx, types.NamespacedName{Name: nodeName}, node)
+			}).Should(BeNotFoundError())
+
+			// In a real world scenario the lease would be deleted by garbage collection because of its owner reference.
+			Consistently(func() bool {
+				Expect(testClient.Get(ctx, types.NamespacedName{Namespace: testNamespace.Name, Name: "gardener-node-agent-" + nodeName}, lease)).To(Succeed())
+				return lease.Spec.RenewTime.Equal(oldRenewTime)
+			}).Should(BeTrue())
+			validateOwnerReference(lease, node)
+		})
+	})
+})
+
+func validateOwnerReference(lease *coordinationv1.Lease, node *corev1.Node) {
+	Expect(lease.OwnerReferences).To(ConsistOf(metav1.OwnerReference{
```

Comment:
```suggestion
	ExpectWithOffset(1, lease.OwnerReferences).To(ConsistOf(metav1.OwnerReference{
```

ref https://github.com/gardener/gardener/blob/master/docs/development/testing.md#writing-tests-relevant-for-all-kinds